PolyEngine/Engine/Src/TransformComponent.cpp

```diff
@@ -0,0 +1,70 @@
+#include "TransformComponent.hpp"
+
+using namespace Poly;
+
+//-----------------------------------------------------------------------------
+void TransformComponent::SetParent(TransformComponent* parent) {
+	Parent = parent;
+	//TODO calculate new LocalTransformation based on parent GlobalTransformation
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalTranslation(const Vector& position) {
+	LocalTranslation = position;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalRotation(const Quaternion& quaternion) { 
+	LocalRotation = quaternion; 
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalScale(const Vector& scale) {
+	LocalScale = scale;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+const Matrix& TransformComponent::GetLocalTransformationMatrix() const {
+	if (Dirty) {
```

Comment:
All bracets should start at the new line according to our coding style. This comment applies to the whole PR.